src/lasdi/latent_space.py

```diff
@@ -130,7 +169,13 @@ def decoder(self, x):
 
         for i in range(1, self.n_layers + 1):
             fc = getattr(self, 'fc' + str(i) + '_d')
-            x = self.g_e(fc(x))
+            x = fc(x) # apply linear layer
+            if hasattr(self, 'a1'): # test if there is at least one attention layer
+                x = x.unsqueeze(1)  # Add sequence dimension for attention
```

Comment:
See the comments above. In this if statement, the inputs would be `x` and `n_layer-i`.